newIDE/app/src/MainFrame/RouterContext.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from 'react';
+import Window, { type AppArguments } from '../Utils/Window';
+
+export type Router = {|
+  routeArguments: AppArguments,
+  removeRouteArguments: (string[]) => void,
+  addRouteArguments: AppArguments => void,
+  navigateToRoute: (route: string, additionalArgument?: AppArguments) => void,
+|};
+
+const initialState: Router = {
+  routeArguments: {},
+  removeRouteArguments: () => {},
+  addRouteArguments: () => {},
+  navigateToRoute: () => {},
+};
+
+const RouterContext = React.createContext<Router>(initialState);
+
+export default RouterContext;
+
+type Props = {|
+  children?: React.Node,
+|};
+
+export const RouterContextProvider = ({ children }: Props) => {
+  const initialWindowArguments = Window.getArguments();
```

Comment:
Might not be worth using a variable for this, you can use `Window.getArguments()` directly in the initial value of the state variable declaration below